compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/window/Dialog.skiko.kt

```diff
@@ -228,88 +229,99 @@ private fun DialogLayout(
                 }
-        }
-    }
+            }
 
-    fun hideDialog() {
-        appearAnimationJob?.cancel()
-        if (ComposeUiFlags.isDialogAnimationEnabled) {
-            CoroutineScope(coroutineContext).launch(start = CoroutineStart.UNDISPATCHED) {
-                val initialProgress = appearanceProgress.value
-                val duration =
-                    durationScale() * initialProgress * AnimatedLayerDisappearanceDuration
-                modifier.value = Modifier.animationLayerTransform(appearanceProgress)
-
-                withAnimationProgress(
-                    duration = duration.seconds,
-                    timingFunction = ::easeOutTimingFunction
-                ) { progress ->
-                    appearanceProgress.value = (1f - progress) * initialProgress
-                    updateScrimLayerColor()
+            private fun updateScrimLayerColor() {
+                layer.scrimColor = scrimColor?.let {
+                    it.copy(it.alpha * contentAlpha(appearanceProgress.value))
                 }
-
-                layer.close()
             }
-        } else {
-            layer.close()
-        }
-    }
-
-    fun updateScrimLayerColor() {
-        val scrimAlpha = contentAlpha(appearanceProgress.value)
-        layer.scrimColor = scrimColor.copy(scrimColor.alpha * scrimAlpha)
-    }
 
-    private fun contentAlpha(progress: Float): Float =
-        AnimatedLayerInitialAlpha + (1f - AnimatedLayerInitialAlpha) * progress
+            private fun contentAlpha(progress: Float): Float =
+                AnimatedLayerInitialAlpha + (1f - AnimatedLayerInitialAlpha) * progress
 
-    private fun durationScale(): Float = coroutineContext[MotionDurationScale]?.scaleFactor ?: 1f
+            private fun durationScale(): Float =
+                coroutineContext[MotionDurationScale]?.scaleFactor ?: 1f
 
-    private fun Modifier.animationLayerTransform(progress: State<Float>): Modifier = graphicsLayer {
-        this.alpha = contentAlpha(progress.value)
-        val reversedProgress = 1f - progress.value
-        val scale = 1f - reversedProgress * AnimatedLayerScale
-        this.scaleX = scale
-        this.scaleY = scale
-        this.translationY = AnimatedLayerOffsetDp * reversedProgress * density
+            private fun Modifier.animationLayerTransform(progress: State<Float>): Modifier =
+                graphicsLayer {
+                    this.alpha = contentAlpha(progress.value)
+                    val reversedProgress = 1f - progress.value
+                    val scale = 1f - reversedProgress * AnimatedLayerScale
+                    this.scaleX = scale
+                    this.scaleY = scale
+                    this.translationY = AnimatedLayerOffsetDp * reversedProgress * density
+                }
+        }
+    } else {
+        object : DialogAnimator {
+            override var scrimColor: Color? by layer::scrimColor
+            override fun showDialog() {}
```

Comment:
The fact that this method is empty is a good hint that `showDialog` is not named correctly. It doesn't actually show the dialog. It will be better if we rename it to, say, `onDialogShown`, and it will also improve a bit the situation with the mismatched location of the show/hide calls.